js.go

```diff
@@ -570,7 +570,12 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, opts []
 	// Check if we are manual ack.
 	if cb != nil && !o.mack {
 		ocb := cb
-		cb = func(m *Msg) { ocb(m); m.Ack() }
+		cb = func(m *Msg) {
+			m.Sub.jsi.autoAck = true
```

Comment:
Can't we just set something once instead of flipping during the callback?

Reply:
Ok changed the implementation a bit so that we only check once in case of using auto ack and async subs.